### ISSUE TITLE
'no-noninteractive-element-to-interactive-role' occurs for <fieldset role="radiogroup" ...> when HTML-ARIA spec states this is a valid role

### ISSUE THREAD

**Issue by meghanking** [jsx-eslint/eslint-plugin-jsx-a11y#774]
Within [HTML-ARIA, the "Rules of ARIA attribute usage by HTML feature" table](https://w3c.github.io/html-aria/#fieldset), fieldset row discusses that allowed roles include "radiogroup".

If this is considered a valid role, why is this linting error occurring?

**Comment by lemnis** [comment]
I just also bumped into this exact issue.

It is related to that it is expected that a radio group allows to change it values with arrows. (see https://www.w3.org/WAI/ARIA/apg/patterns/radiobutton/). 

But I personally believe this keyboard accessibility is not mandatory.

**Comment by ljharb** [comment]
It is quite mandatory; I use it all the time.

**Comment by lemnis** [comment]
Than close this ticket with a explanation 😉

**Comment by dietergeerts** [comment]
From MDN (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/radiogroup_role):

> In this [<fieldset>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) example, while role="radiogroup" is not necessary, to have this grouping explicitly announced as radiogroup, include the ARIA role.

So that would mean it should be allowed, if not, I would suspect such major site on standards must be updated to reflect this.

**Comment by jessebeach** [comment]
@dietergeerts interesting case!

Putting `role="radiogroup"` on the `fieldset` element would erase the fieldset semantics. Rather than applying the role to the `fieldset` element, I would recommend wrapping the contents of the fieldset in a `div` with the role `radiogroup`, as a sibling element to the `legend` element.

For example: https://codepen.io/jessebeach/pen/LYmrRmz
<img width="1438" alt="Screen Shot 2022-10-03 at 10 04 38 AM" src="https://user-images.githubusercontent.com/345913/193636563-6f28699d-74bd-4847-8502-0bd61ef6624b.png">

**Comment by jessebeach** [comment]
So this rule is correct in disallowing the application of `radiogroup` to a `fieldset` element.

**Comment by ljharb** [comment]
Sounds like this can be closed, thanks for the investigation!

**Comment by Georgegriff** [comment]
I'm a little confused at the resolution here if it's in the spec https://w3c.github.io/html-aria/#fieldset shouldn't the linter do what the spec says?
![image](https://user-images.githubusercontent.com/9056958/195896461-6a9a819c-f68e-48a2-a4c7-f90a98833309.png)

**Comment by jessebeach** [comment]
I share @ljharb's perspective here. This linter should encourage a great user experience. Specs also try to do this, but they are very slow to evolve (ARIA 1.2 has been in draft for years) and sometimes their recommendations don't pan out in practice (which is understandable). So we try to recommend the practice that produces a great user experience, informed by the specs. I also feed what we learn back to the spec authors, so that we can improve the next versions of the specs!

**Comment by Georgegriff** [comment]
Thank you both the reply! 

I think i'm getting it, and I think I may understand now. 

If i'm reading it correctly fieldset provides the semantics for the "legend" labelling the radiogroup?

I'm my instance i'm not in control of the the labelling so i've established an aria-describedby + aria-labelledby relationship instead like so:
In my screen reader tests this seems to work as i expect, i think.
```html
<p id="my-label">Whats the number?</p>
<fieldset aria-labelledby="my-label">
  <label>  
  <input type="radio" id="role1" value="1" name="role1">
  One
</label>
<label>
   <input type="radio" id="role2" value="2" name="role2" >
   Two
   </label>
</fieldset>
```

But i think if i'm understanding what you are saying in this instance my fieldset could just be a `<div role="radiogroup` instead because i don't require fieldset semantics for labelling due to my aria relationship?

**Comment by dietergeerts** [comment]
@ljharb , @jessebeach,

Thx for the reply!

I see now that what the specs says is possible, but considering UX, it's not the best way to do it. And this library tries to "enforce" a better UX, which I think is a good thing. That said, it would be very helpful to have a note in the documentation of the rule to state this, because when you encounter this, and you normally go by the spec to implement your code, you have no clue why this linter is complaining. So maybe this ticket can be reopened as work to update the docs of the rule?

**Comment by jessebeach** [comment]
> So maybe this ticket can be reopened as work to update the docs of the rule?

@dietergeerts that's a fair resolution
